frontend/packages/ceph-storage-plugin/src/utils/dashboard.ts

```diff
@@ -1,11 +1,11 @@
 import * as _ from 'lodash';
 import { PrometheusResponse } from '@console/internal/components/graphs';
 
-export const getResiliencyProgress = (results: PrometheusResponse): number => {
+export const getResiliencyProgress = (results: PrometheusResponse): string => {
   const progress = _.get(results, 'data.result[0].value[1]');
-  //  Zero PG Case
-  if (Number.isNaN(progress)) {
-    return null;
+  const convertedProgress = Number(progress);
+  if (Number.isNaN(convertedProgress)) {
+    return undefined;
```

Comment:
the function returns a string and we are returning undefined?

Reply:
By default null and undefined are subtypes of all other types

Reply:
Ref : https://www.typescriptlang.org/docs/handbook/basic-types.html#null-and-undefined